### PR TITLE
[AKS] Fix for app routing CLI

### PR DESCRIPTION
**Pull request by bosesuneha**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

aks create -g {resource_group} -n {cluster_name} --enable-addons web_application_routing

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -3081,8 +3081,10 @@ def construct_mc_profile_preview(self, bypass_restore_defaults: bool = False) ->
         mc = self.set_up_image_integrity(mc)
         # set up cluster snapshot
         mc = self.set_up_creationdata_of_cluster_snapshot(mc)
-        # set up app routing profile
+        # set up app routing profile using `aks approuting enable`
         mc = self.set_up_app_routing_profile(mc)
+        #set up app routing profile using `aks enable-addons`
+        mc = self.set_up_ingress_web_app_routing(mc)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3087]: Is it possible to merge these functions into one?

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -3081,8 +3081,10 @@ def construct_mc_profile_preview(self, bypass_restore_defaults: bool = False) ->
         mc = self.set_up_image_integrity(mc)
         # set up cluster snapshot
         mc = self.set_up_creationdata_of_cluster_snapshot(mc)
-        # set up app routing profile
+        # set up app routing profile using `aks approuting enable`
         mc = self.set_up_app_routing_profile(mc)
+        #set up app routing profile using `aks enable-addons`
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3086]:
```suggestion
        #set up app routing profile using `aks create --enable-addons web_application_routing`
```
the implementation of command `aks enable-addons` is in `custom.py`

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
Comment (bosesuneha) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: merged them into one function

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=82663665&view=results)

- test_aks_create_with_web_application_routing 
- test_aks_create_and_update_web_application_routing_dns_zone 
- test_aks_disable_addon_web_app_routing 
- test_aks_create_web_application_routing_with_private_dns_zone 
- test_aks_create_with_app_routing_enabled 
- test_aks_approuting_enable_disable 
- test_aks_approuting_enable_with_keyvault_secrets_provider_addon 
- test_aks_approuting_update 
- test_aks_approuting_zone_add_delete_list 
- test_aks_approuting_zone_update

**Comment by FumingZhang** [comment]
Please left a note in history

**Review by FumingZhang — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]

**Review by FumingZhang — COMMENTED** [review]
SDK/default API version updated, please rerun live test and commit new recording file

https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=83092072&view=results

**Comment by yanzhudd** [comment]
Please add changlog to the HISTORY.rst file

File: src/aks-preview/HISTORY.rst [hunk 3]
```diff
@@ -14,6 +14,10 @@ Pending
 * Vendor new SDK and bump API version to 2023-10-02-preview.
 * Update the enum for `--os-sku` in command `az aks nodepool update` to only accept the expected Ubuntu and AzureLinux OSSKUs.
 
+0.5.172
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:17]: please update `VERSION` in setup.py at the same time if you want to release a new version of aks-preview AND move all those notes under the pending section (several lines above) to this block.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 4]
```diff
@@ -6398,28 +6398,34 @@ def test_aks_create_with_web_application_routing(self, resource_group, resource_
 
     @AllowLargeResponse()
     @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
-    def test_aks_create_web_application_routing_dns_zone_not_exist(self, resource_group, resource_group_location):
+    def test_aks_create_web_application_routing_with_private_dns_zone(self, resource_group, resource_group_location):
```
Comment (yanzhudd) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:6401]: Let me check with you if the origina test name 'test_aks_create_web_application_routing_dns_zone_not_exist' already not exists. If so, please check if the recording file test_aks_create_web_application_routing_dns_zone_not_exist.yaml has been removed.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**